src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSource.java

```diff
@@ -103,14 +101,21 @@ public void prepareRun(BatchSourceContext context) {
     Configuration conf = null;
     try {
       conf = getConfiguration(collector);
-      validateOutputSchema(conf, collector);
+
     } catch (IOException e) {
       collector.addFailure(
         String.format("Failed to prepare configuration for job : %s", e.getMessage()), null)
         .withConfigProperty(BigtableSourceConfig.BIGTABLE_OPTIONS)
         .withStacktrace(e.getStackTrace());
+      collector.getOrThrowException();
+    }
+    try {
+      validateOutputSchema(conf, collector);
+    } catch (IOException e) {
+      collector.addFailure(String.format("Failed to connect to Bigtable : %s", e.getMessage()), null)
```

Comment:
why is this not a warning?

Reply:
It still needs to fail during pipeline run, just not during pipeline deployment